kubectl-plugin/pkg/util/generation/generation.go

```diff
@@ -26,40 +36,67 @@ type RayClusterSpecObject struct {
 	RayVersion *string `yaml:"ray-version,omitempty"`
 	Image      *string `yaml:"image,omitempty"`
 
-	HeadCPU              *string           `yaml:"head-cpu,omitempty"`
-	HeadGPU              *string           `yaml:"head-gpu,omitempty"`
-	HeadMemory           *string           `yaml:"head-memory,omitempty"`
-	HeadEphemeralStorage *string           `yaml:"head-ephemeral-storage,omitempty"`
-	HeadRayStartParams   map[string]string `yaml:"head-ray-start-params,omitempty"`
-	HeadNodeSelectors    map[string]string `yaml:"head-node-selectors,omitempty"`
+	Head *Head `yaml:"head,omitempty"`
+
+	GKE *GKE `yaml:"gke,omitempty"`
+
+	WorkerGroups []WorkerGroup `yaml:"worker-groups,omitempty"`
+}
+
+type Head struct {
+	CPU              *string           `yaml:"cpu,omitempty"`
+	GPU              *string           `yaml:"gpu,omitempty"`
+	Memory           *string           `yaml:"memory,omitempty"`
+	EphemeralStorage *string           `yaml:"ephemeral-storage,omitempty"`
+	RayStartParams   map[string]string `yaml:"ray-start-params,omitempty"`
+	NodeSelectors    map[string]string `yaml:"node-selectors,omitempty"`
+}
 
-	WorkerGroups []WorkerGroupConfig `yaml:"worker-groups,omitempty"`
+type WorkerGroup struct {
+	Name             *string           `yaml:"name,omitempty"`
+	CPU              *string           `yaml:"cpu,omitempty"`
+	GPU              *string           `yaml:"gpu,omitempty"`
+	Memory           *string           `yaml:"memory,omitempty"`
+	EphemeralStorage *string           `yaml:"ephemeral-storage,omitempty"`
+	RayStartParams   map[string]string `yaml:"ray-start-params,omitempty"`
+	NodeSelectors    map[string]string `yaml:"node-selectors,omitempty"`
+	Replicas         int32             `yaml:"replicas"`
 }
```

Comment:
Do you consider to use embeded struct and `yaml:",inline"` for these 2 structs? Because the fields are almost the same.

Reply:
good idea. I tried to see how it looks [here](https://github.com/davidxia/kuberay/pull/4/files#diff-d7990c8ae1278889923e749905d9699559a9993a95a6bb7e7b62cca21968c3c7). It gets dryer here in the struct definition. But it actually results in ~50 more lines of code because we have to add more lines to set the attributes now.